File: M1/azor-chatdog-py/src/assistant/registry.py
from .assistent import Assistant

class AssistantRegistry:
    _assistants: dict[str, Assistant] = {}
    
    @classmethod
    def register(cls, assistant: Assistant):
        cls._assistants[assistant.id] = assistant
    
    @classmethod
    def get(cls, assistant_id: str) -> Assistant | None:
        return cls._assistants.get(assistant_id)
    
    @classmethod
    def list_all(cls) -> list[Assistant]:
        return list(cls._assistants.values())
    
    @classmethod
    def get_default(cls) -> Assistant:
        return cls._assistants.get('azor')

File: M1/azor-chatdog-js/src/assistant/registry.ts
import { Assistant } from "./assistant.js";
import { createAzaAssistant } from "./aza.js";
import { createReksioAssistant } from "./reksio.js";
import { createAzorAssistant } from "./azor.js";

export class AssistantRegistry {
    private static assistants: Map<string, Assistant> = new Map();

    public static register(assistant: Assistant): void {
        this.assistants.set(assistant.id, assistant);
    }

    public static get(id: string): Assistant | undefined {
        return this.assistants.get(id);
    }

    public static list(): Assistant[] {
        return Array.from(this.assistants.values());
    }
    
    public static has(id: string): boolean {
        return this.assistants.has(id);
    }

    public static registerBuiltinAssistants(): void {
        this.register(createAzaAssistant());
        this.register(createReksioAssistant());
        this.register(createAzorAssistant());
    }
}